File: Pset2/caesar.c
#include<cs50.h>
#include<stdio.h>
#include<ctype.h>

int main(int argc, string argv[])
{
    if(argc==2){
        string f = argv[1];
        int k = atoi(f);
        printf("plaintext: ");
        string s = get_string();

        for(int n=0; s[n]!='\0'; n++)
        {
            if(isalpha(s[n])){
                if(isupper(s[n]))
                    s[n] = (s[n] -65 + k) % 26 + 65;
                else if(islower(s[n]))
                    s[n] = (s[n] -97 + k) % 26 + 97;
            }
        }

        printf("ciphertext: %s\n", s);
    }
    else{
        eprintf("Usage: ./caesar k\n");
        return 1;
    }

}

File: PSet6/caesar/caesar.py
from cs50 import get_string
import sys

def main():
    if len(sys.argv) == 2:
        f = sys.argv[1]
        k = int(f)
        print("plaintext: ", end="")
        s = get_string()
        print("ciphertext: ", end="")
        for c in s:
            #checking if alphabet
            if c.isalpha():
                #checking if uppercase
                if c.isupper():
                    d = (ord(c) - 65 + k) % 26 + 65
                    c = chr(d)
                    print(c, end="")
                #checking if lowercase
                elif c.islower():
                    d = (ord(c) - 97 + k) % 26 + 97
                    c = chr(d)
                    print(c, end="")
            else:
                print(c, end="")
        print("")

    else:
        print("Usage: ./caesar k")
        sys.exit(1)

if __name__ == "__main__":
    main()